cc/testing/evaluation/README.md

```diff
@@ -0,0 +1,91 @@
+
+The testing algorithms have been engineered to violate differential privacy by manipulating the epsilon value. Under typical circumstances, the epsilon value in a differentially private algorithm determines the amount of noise applied to the data. Smaller epsilon values result in more noise, which leads to greater privacy protection. Larger epsilon values result in less noise, which leads to less privacy protection. The testing tools use this epsilon value (along with other parameters) to assess if the amount of noise that has been applied to the data is consistent with the mathematical definition of differential privacy.
+
+However, if the testing tools evaluate a given algorithm using an epsilon value that is smaller than the epsilon value used to generate noise, then a differential privacy violation occurs. This is because `insufficient_noise` has been applied to the algorithm; in other words, the algorithm is claiming more privacy protection than it actually provides.
+
+For all algorithms with `insufficient_noise`, the `advertised_epsilon`, or the epsilon value used to evaluate the algorithm for differential privacy, *is always less than* the `implemented_epsilon`, or the actual amount of noise applied to the algorithm. This deliberate mismatch ensures that the algorithm will *always* violate differential privacy, because it is always claiming to have more privacy protection than the true amount of noise applied to the data.
+
+The relationship between the `advertised_epsilon` and the `implemented_epsilon` can be represented by a third variable, `ratio`, where `ratio = advertised_epsilon / implemented epsilon.` For example, if `advertised_epsilon = 2` and `implemented_epsilon = 5`, then `ratio = 0.4`, meaning that only 40% of privacy protection that is claimed is actually being applied.
+
+## How to Use
+
+The evaluation is run in two parts. Part 1 is written in C++ and Part 2 is written in Java. Part 1 must be run before Part 2.
+
+### Part 1
+
+#### Testing the Stochastic Tester
+
+The first stage of Part 1 runs the Stochastic Tester on three algorithms with `insufficient noise` (Count, BoundedSum, and BoundedMean) over a continuous range of ratio values. Given that the algorithms are known to violate differential privacy, the desired outcome for every test is `false`, meaning that the algorithm *does not* satisfy differential privacy. 
+
+Note that the Stochastic Tester test contains several default parameters:
+
+	num_samples_per_histogram = 100
+	ratio_min = 80.0
+	ratio_max = 85.0
```

Comment:
SG. I'm going to set the ratio_min at 0.50, err on the conservative side since Stochastic Tester only achieved a ratio max in the high-sixties, and future algorithms might be even lower.